### PR TITLE
dom0: Add support for Zephyr OS as guest

### PR DESCRIPTION
**Pull request by rshym**

### CODE REVIEW

File: meta-xt-control-domain/recipes-guest/domz/files/domz-generic.cfg [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+# Kernel image to boot
+kernel="/usr/lib/xen/boot/zephyr-domz"
+
+# Number of VCPUS
+vcpus=1
+
+# Initial memory allocation (MB)
+memory=16
+
+gic_version="v2"
```
Comment (lorc) [inline meta-xt-control-domain/recipes-guest/domz/files/domz-generic.cfg:19]: Are you sure that Zephyr does not support GICv3? Please mind that this is `meta-xt-common` layer, which is used by all platforms.

Comment (rshym) [reply meta-xt-control-domain/recipes-guest/domz/files/domz-generic.cfg:19]: I just used the first mentioned config from https://docs.zephyrproject.org/latest/boards/arm64/xenvm/doc/index.html.

As I understand GIC version should be specified in this config line and in board `west build -b xenvm_gicv3 ...`.
Seem that I need to pass `board` variable from yaml down to the recipe to select the proper config.

**Comment by rshym** [comment]
PR requires https://github.com/xen-troops/meta-xt-prod-devel-rcar/pull/44 to be merged.
Tested successfully on KF.
Image was built by following commands:
```
moulin ../prod-devel-rcar.yaml --MACHINE h3ulcb-4x2g-kf --ENABLE_ZEPHYR yes
ninja image-full
```
After flash to eMMC, a successful wark of Zephyr as guest was observed:
```
root@generic-armv8-xt-dom0:~# xl list
Name                                        ID   Mem VCPUs      State   Time(s)
Domain-0                                     0   256     4     r-----       9.4
DomZ                                         1    16     1     -b----       1.5
DomD                                         2  2047     4     r-----      17.5
root@generic-armv8-xt-dom0:~# xl console DomZ
*** Booting Zephyr OS build zephyr-v3.1.0-2635-g1dd76ac08cdf  ***
thread_a: Hello World from cpu 0 on xenvm!
[00:00:00.071,000] <inf> xen_events: xen_events_init: events inited

[00:00:00.071,000] <inf> uart_hvc_xen: Xen HVC inited successfully

thread_b: Hello World from cpu 0 on xenvm!
thread_a: Hello World from cpu 0 on xenvm!
...
```

File: meta-xt-control-domain/recipes-guest/domz/files/domz-generic.cfg (hunk 1 above)
Comment (rshym) [reply meta-xt-control-domain/recipes-guest/domz/files/domz-generic.cfg]: `gic_version` is set to `default` and successfully tested on HW with GIC v2.

**Comment by rshym** [comment]
Ready for re-review.

**Comment by lorc** [comment]
`Acked-by: Volodymyr Babchuk <volodymyr_babchuk@epam.com>`

**Comment by rshym** [comment]
`Ack` is added.
Tested successfully, see logs in comment above.
Ready to be merged.

**Pull request merged**